artifacts/src/main/java/org/openmbee/sdvc/artifacts/service/DefaultArtifactService.java

```diff
@@ -0,0 +1,211 @@
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.*;
+
+@Service
+public class DefaultArtifactService implements ArtifactService {
+
+    private ArtifactStorage artifactStorage;
+    private ServiceFactory serviceFactory;
+    private ProjectDAO projectRepository;
+
+    @Autowired
+    public void setArtifactStorage(ArtifactStorage artifactStorage) {
+        this.artifactStorage = artifactStorage;
+    }
+
+    @Autowired
+    public void setServiceFactory(ServiceFactory serviceFactory) {
+        this.serviceFactory = serviceFactory;
+    }
+
+    @Autowired
+    public void setProjectRepository(ProjectDAO projectRepository) {
+        this.projectRepository = projectRepository;
+    }
+
+    @Override
+    public ArtifactResponse get(String projectId, String refId, String id, Map<String, String> params) {
+        NodeService nodeService = getNodeService(projectId);
+        ElementJson elementJson = getElement(nodeService, projectId, refId, id, params);
+        if(elementJson == null){
+            return null;
+        }
+
+        ArtifactJson artifact = getExistingArtifact(ArtifactJson.getArtifacts(elementJson), params);
+        if(artifact != null) {
+            byte[] data = artifactStorage.get(artifact.getLocation(), elementJson, artifact.getMimeType());
+            ArtifactResponse response = new ArtifactResponse();
+            response.setData(data);
+            response.setExtension(artifact.getExtension());
+            response.setMimeType(artifact.getMimeType());
+            return response;
+        }
+        return null;
+    }
+
+    @Override
+    public ElementsResponse createOrUpdate(String projectId, String refId, String id, MultipartFile file, String user, Map<String, String> params) {
+        NodeService nodeService = getNodeService(projectId);
+        ElementJson elementJson = getElement(nodeService, projectId, refId, id, params);
+
+        if(elementJson == null) {
+            elementJson = new ElementJson();
+            elementJson.setProjectId(projectId);
+            elementJson.setId(id);
+            elementJson.setInRefIds(Arrays.asList(refId));
+        }
+
+        byte[] fileContents;
+        try {
+            fileContents = file.getBytes();
+        } catch (IOException e) {
+            throw new BadRequestException("Could not get contents of multipart file.");
+        }
+
+        String mimeType = getMimeTypeOfFile(file);
+        String fileExtension = getFileExtension(file);
+        String artifactLocation = artifactStorage.store(fileContents, elementJson, mimeType);
+
+        elementJson = attachOrUpdateArtifact(elementJson, artifactLocation, fileExtension, mimeType, "internal");
+        ElementsRequest elementsRequest = new ElementsRequest();
+        elementsRequest.setElements(Arrays.asList(elementJson));
+        nodeService.createOrUpdate(projectId, refId, elementsRequest, params, user);
```

Comment:
can just return the response from nodeService, there's no guarantee the elementJson is updated in place with all the new metadata